astropy/utils/decorators.py

```diff
@@ -96,10 +100,10 @@ def get_function(func):
             func = func.__func__
         return func
 
-    def deprecate_function(func, message):
+    def deprecate_function(func, message, warning_type):
```

Comment:
Adding a positional argument is breaking backward compatibility. Since this is not a "hidden" method, I am not very comfortable with this. Maybe make this an optional keyword instead?

*This comment also applies to all similar code changes that follow.*

Reply:
actually it's not exposed to anywhere in the docs, and neither used by any package: http://docs.astropy.org/en/stable/utils/index.html#id5

but either case, it can be a kwarg

Reply:
So, you don't think someone out there might be using this for whatever reason? It is not "hidden" after all. Depends on how strict you are on the "no API change in LTS" rule.

Reply:
how would you define hidden? It's not exposed to the API docs, and that was my definition rather than having a leading "_" in the name (that does nothing in return).

Reply:
But either case, I'll make it a kwarg, just got a few more urgent fires to tend to first 😅